src/data/httpModule.js

```diff
@@ -82,15 +91,123 @@ export default {
+                    logStatus.localChange = false;
+                  }
+                }
+              }
+            });
+            return logStatus;
+          }
+          // Return all assets/ areas associated with logs
+          function getAttached(log, attribute, resources, resId) {
+            // Only get attached if that attrib exists.  Some logs have no areas!
+            if (log[attribute]) {
+              const logAttached = [];
+              resources.forEach((resrc) => {
+                log[attribute].forEach((attrib) => {
+                  if (resrc[resId] === attrib.id) {
+                    logAttached.push(resrc);
+                  }
+                });
+              });
+              return logAttached;
+            }
+          }
+          // Process each log on its way from the server to the logFactory
+          function processLog(log) {
+            const allAreas = rootState.farm.areas;
+            const allAssets = rootState.farm.assets;
+            const checkStatus = checkLog(log);
+            const attachedAssets = getAttached(log, 'field_farm_asset', allAssets, 'id');
+            const attachedAreas = getAttached(log, 'field_farm_area', allAreas, 'tid');
+            // If the log is not present locally, add it.
+            // If the log is present locally, but has not been changed since the last sync,
+            // update it with the new version from the server
+            // If the log is present locally and has been changed, do not update it.
+            if (checkStatus.localId === null) {
+              console.log('ADDING LOG WITH PARAMS: ', log);
+              commit('addLogFromServer',
+                logFactory({
+                  ...log,
+                  wasPushedToServer: true,
+                  field_farm_area: attachedAreas,
+                  field_farm_asset: attachedAssets,
+                }, STOREFROMSERVER));
+            } else if (!checkStatus.localChange) {
+              // Update the log with all data from the server
+              console.log (`UPDATING UNCHANGED LOG ${log.name}`);
+              const updateParams = {
+                index: checkStatus.storeIndex,
+                log: logFactory({
+                  ...log,
+                  wasPushedToServer: true,
+                  local_id: checkStatus.localId,
+                  field_farm_area: attachedAreas,
+                  field_farm_asset: attachedAssets
+                }, STOREFROMSERVER)
+              }
+              commit('updateLogFromServer', updateParams)
+            } else {
+              console.log(`LOG ${log.name} HAS BEEN CHANGED LOCALLY; WILL NOT BE UPDATED FROM THE SERVER`);
+            }
+          }
+          // Process one or more logs
+          if (res.list) {
+            res.list.forEach(log => processLog(log));
+          } else {
+            processLog(res);
+          }
+        })
+        .catch((err) => { throw err; });
+      // Errors are handled in index.js
```

Comment:
Let's discuss this... I feel like maybe this should all be handled in one place.

Reply:
We are no longer going to append the full area and asset objects to logs received from the server using getAttached.  Instead, we are going to reference areas and assets in our logs in the local store, using the same format as the server's response to getLogs.  The getAttached logic will be moved to editLog, where it will be used to access the areas and assets referenced in the individual log being edited.  This should improve app performance, and reduce redundancy in the app store.